src_c/line.c

```diff
@@ -490,6 +490,89 @@ pg_line_flip_ip(pgLineObject *self, PyObject *_null)
     Py_RETURN_NONE;
 }
 
+double 
+_lerp_helper(float start, float end, float amount) {
+    return (1 - amount) * start + amount * end;
+}
+
+static int
+_line_scale_helper(pgLineObject *line, double factor, double origin) {
+    if (factor == 1.0) {
+        return 1;
+    }
+    else if (factor == 0.0) {
+        PyErr_SetString(PyExc_ValueError, "Cannot scale by 0");
+        return 0;
+    }
+    else if (factor < 0.0) {
+        PyErr_SetString(PyExc_ValueError, "Cannot scale by a negative number");
+        return 0;
+    }
```

Comment:
these two could be condensed to a single one. The error would just say that you can only scale by a positive non-zero number.